src/Translator/Extractor/AdminExtractor.php

```diff
@@ -127,6 +127,16 @@ public function setPrefix($prefix): void
 
     public function getLabel(string $label, string $context = '', string $type = ''): string
     {
+        if (null === $this->catalogue) {
+            throw new \LogicException('The catalogue is not set.');
+        }
+        if (null === $this->labelStrategy) {
+            throw new \LogicException('The label strategy is not set.');
+        }
+        if (null === $this->domain) {
+            throw new \LogicException('The domain is not set.');
```

Comment:
is there anything the user could do to avoid this exceptions, like calling some other method?

Reply:
The user shouldn't use this method.

The extractor is doing the following logic
```
$admin = $this->adminPool->getInstance($id);

            $this->labelStrategy = $admin->getLabelTranslatorStrategy();
            $this->domain = $admin->getTranslationDomain();

            $label = $admin->getLabel();
            if (!empty($label)) {
                $catalogue->set($label, $this->prefix.$label, $admin->getTranslationDomain());
            }

            $admin->setLabelTranslatorStrategy($this);
            $admin->setSubject($admin->getNewInstance());

            foreach (self::PUBLIC_ADMIN_METHODS as $method) {
                $admin->$method();
            }

            foreach (self::BREADCRUMB_ACTIONS as $action) {
                $this->breadcrumbsBuilder->getBreadcrumbs($admin, $action);
            }
```

The `getLabel` will be called by the admin automatically but should never be called manually by the user.